server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -441,6 +455,11 @@ public void addDynamicMapper(Mapper mapper) {
         }
 
         void postParse() {
+            if (dataStreamTimestampField != null) {
```

Comment:
Small comment, can collapse these two 'if' checks.

Also, same thought as above about including 'data stream' and the field name in the error message.